packages/block-editor/src/components/block-list/index.js

```diff
@@ -151,29 +151,51 @@ export default function BlockList( settings ) {
 function Items( {
 	placeholder,
 	rootClientId,
-	renderAppender,
+	renderAppender: CustomAppender,
 	__experimentalAppenderTagName,
 	layout = defaultLayout,
 } ) {
-	const { order, selectedBlocks, visibleBlocks, temporarilyEditingAsBlocks } =
-		useSelect(
-			( select ) => {
-				const {
-					getBlockOrder,
-					getSelectedBlockClientIds,
-					__unstableGetVisibleBlocks,
-					__unstableGetTemporarilyEditingAsBlocks,
-				} = select( blockEditorStore );
-				return {
-					order: getBlockOrder( rootClientId ),
-					selectedBlocks: getSelectedBlockClientIds(),
-					visibleBlocks: __unstableGetVisibleBlocks(),
-					temporarilyEditingAsBlocks:
-						__unstableGetTemporarilyEditingAsBlocks(),
-				};
-			},
-			[ rootClientId ]
-		);
+	// Avoid passing CustomAppender to useSelect because it could be a new
+	// function on every render.
+	const hasAppender = CustomAppender !== false;
+	const hasCustomAppender = !! CustomAppender;
```

Comment:
What's the difference between these two flags?

Reply:
I hoped the cost name was descriptive 😄
One is to check if there's an appender in general: either CustomAppender === undefined or a component
The other is to check if there's a custom appender defined: CustomAppender must be a function (truthy)

Reply:
I guess the "undefined" is what tricked me

Reply:
I guess we should set a default to true

Reply:
Actually that won't work because we have another check CustomAppender ? ...

Reply:
I guess we could check boolean there, anyway I left it as is for now.